### PR TITLE
time/span: add / and * methods

### PR DESCRIPTION
**Pull request by jbaum98**
Allows you to multiplay and divide a `Time::Span` by a number.

### CODE REVIEW

File: src/time/span.cr [hunk 1]
```diff
@@ -214,6 +214,24 @@ struct Time::Span
...
   end
 
+  def *(number : Number)
+    # TODO check overflow
+    Span.new(ticks * number)
+  end
+
+  def *
+    self
+  end
```
Comment (jhass) [inline src/time/span.cr]: I don't think we have a syntactic way to call these overloads?

File: src/time/span.cr [hunk 2]
```diff
@@ -214,6 +214,24 @@ struct Time::Span
...
+
+  def /(number : Number)
+    # TODO check overflow
+    Span.new(ticks / number)
+  end
+
+  def /
+    self
+  end
```
Comment (jhass) [inline src/time/span.cr]: Same

**Comment by asterite** [comment]
In which scenario you'd need to multiply or divide a time span?

**Comment by jhass** [comment]
I struggled first too, but I can see it's useful. Let's say you're splitting or merging video files for example, you have the duration and need the duration of the target (for display or whatever).

```cr
snippet_duration = duration / pieces
```

**Comment by ozra** [comment]
I for one multiply time spans in code (have my own implementation still, but soon it might be ditchable :) )

**Comment by jbaum98** [comment]
I added this in part to make it more compatible with Ruby because Ruby treats times like raw numbers.
```ruby
x = Time.now
sleep 4
y = Time.now
diff = y - x #=> A float representing the number of seconds between y and x, so approx 4
diff * 2     # Just regular arithmetic
```

**Comment by jbaum98** [comment]
I added the versions that don't take arguments following the example of addition and subtraction, but I'm not sure we need them. Why are they there for addition and subtraction?

**Comment by jhass** [comment]
The "addition" and "substraction" overloads are actually the prefix operators, `+@` and `-@` in Ruby respectively. They get called when you do `a = +duration` or `a = -duration` for example.

**Comment by jbaum98** [comment]
Oh cool. So they're unnecessary for multiplication/division?

**Comment by jhass** [comment]
Since @asterite didn't respond with any counter arguments, let's add this.

**Pull request merged**